toolkit/tools/imagegen/installutils/overlay.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package installutils
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"microsoft.com/pkggen/imagegen/diskutils"
+	"microsoft.com/pkggen/internal/logger"
+)
+
+// Overlay Struct representing an overlay mount
+type Overlay struct {
+	DevicePath string
+}
+
+func (o Overlay) getMountArgs() string {
+
+	_, deviceName := filepath.Split(o.DevicePath)
```

Comment:
Minor: It looks like these path calculations are done in most of the routines here. I'd recommend creating an initialization routine for this struct (something like `NewOverlay(...)`) that calculates these values and stores them in private fields in the struct.

Reply:
Fixed.

---
In reply to: [574705362](https://github.com/microsoft/CBL-Mariner/pull/611#discussion_r574705362) [](ancestors = 574705362)